contracts/contracts/solidity/mocks/MockStarknetMessaging.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: Apache-2.0.
+pragma solidity >0.6.0 <=0.8.0;
+
+import '../../../vendor/starkgate-contracts-solidity-v0.8/src/starkware/starknet/solidity/StarknetMessaging.sol';
```

Comment:
I had to create this file because starkgate is using a different path for NamedStorage.sol which results in a HH404: File not found error when the solidity compiler tries to retrieve this files. https://github.com/starkware-libs/starkgate-contracts/blob/c08863a1f08226c09f1d0748124192e848d73db9/src/starkware/starknet/solidity/StarknetMessaging.sol